testsuite/tests/basic/eval_order_6.ml

```diff
@@ -0,0 +1,13 @@
+let f { contents = c } =
+    fun () -> c;;
+
+let r = ref 10;;
+
+let h = f r;;
+
+print_endline (string_of_int (h ()));;
+
+r := 20;;
+
+print_endline (string_of_int (h ()));;
+
```

Comment:
I suggest to extend the test to cover the case of a record with some mutable and some immutable fields.